File: backend/app/Models/Evaluator.php
<?php

namespace App\Models;

use App\Enum\ProjectTypeEnum;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Evaluator extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'user_id',
        'PIN',
    ];

    public function assessments()
    {
        return $this->hasMany(Assessment::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function categories()
    {
        return $this->belongsToMany(Category::class, 'evaluator_categories');
    }

    public static function generateRandomPin() {
        do {
            $pin = rand(1111, 9999);

            $pinExists = self::where('PIN', $pin)->exists();
        } while ($pinExists);

        return $pin;
    }

    public function getTotalProjectsAttribute() {
        return $this->assessments->count();
    }
}

File: backend/app/models/evaluator.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base

class Evaluator(Base):
    __tablename__ = "evaluators"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    PIN = Column(String(4), unique=True, nullable=False)
    year = Column(Integer, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    deleted_at = Column(DateTime(timezone=True), nullable=True)
    
    user = relationship(
        "User",
        back_populates="evaluator",
        primaryjoin="and_(User.id==Evaluator.user_id, User.deleted_at==None)"
    )

    assessments = relationship(
        "Assessment",
        back_populates="evaluator",
        primaryjoin="and_(Assessment.evaluator_id==Evaluator.id, Assessment.deleted_at==None)"
    )

    categories = relationship(
        "Category",
        secondary="evaluator_categories",
        back_populates="evaluators",
        primaryjoin="and_(Evaluator.id==evaluator_categories.c.evaluator_id, Evaluator.deleted_at==None)",
        secondaryjoin="and_(Category.id==evaluator_categories.c.category_id, Category.deleted_at==None)"
    )
        
    @property
    def total_projects(self) -> int:
        return len(self.assessments) 